core/src/main/java/tc/oc/pgm/destroyable/DestroyableFactory.java

```diff
@@ -65,8 +90,12 @@ public boolean getShowProgress() {
     return this.showProgress;
   }
 
-  public boolean hasSparks() {
-    return this.sparks;
+  public boolean isSparksActive() {
+    return this.sparks != SparksType.SPARKS_NONE;
+  }
+
+  public boolean isSparksAll() {
+    return this.sparks != SparksType.SPARKS_NEAR;
```

Comment:
It would return true if you have set this to sparks none, it kinda makes no sense

```suggestion
    return this.sparks == SparksType.ALL;
```